docs/source/performance.rst

```diff
@@ -0,0 +1,154 @@
+.. _performance:
+
+Performance indicators
+----------------------
+
+In the upcoming sections, you can find information on how to write performance
+indicators, how to incorporate them into Thoth and how to inject them into
+Thoth's recommendation pipeline.
+
+Writing a performance script
+============================
+
+Performance related characteristics are automatically gathered on Thoth's
+execution part - `Amun <https://github.com/thoth-station/amun-api>`_ which can
+be triggered directly or using :ref:`dependency_monkey`. Amun accepts
+`specification` which is turned into a build and subsequent job which verifies
+the given software stack in the given runtime environment as described in the
+specification.
+
+The performance script which is supplied to Amun should be directly executable
+(e.g. ``python3 script.py``), it can print additional information onto
+``stderr`` in any form (this output is captured by Amun for additional
+analysis). The output written onto ``stdout`` *has to be* in a JSON format with
+any keys and values the script wants to capture and required ``@parameters``
+and ``@result`` keys (see bellow). Optional ``overall_score`` key states
+"overall score" of the performance indicator.
+
+The script *has to report* following keys to stdout in the resulting JSON:
+
+* ``@parameters`` (type ``dict``) - parameters which define the given performance script (e.g. matrix size in case of matrix multiplication)
+* ``@result`` (type ``dict``) - the actual result which was obtained during the performance indicator run
+* ``name`` (type ``str``) - the name of performance indicator, this name has to match graph database model in thoth-storages (see bellow)
+* ``framework`` (type ``str``) - name of the tested framework (e.g. ``tensorflow``, ``pytorch``, ...) all in lowercase (should conform to the package name)
+* ``tensorflow_buildinfo`` (type ``dict``) - in case of TensorFlow performance indicator, the value under this key holds automatically gathered build information of the tested TensorFlow wheel file (available in `AICoE optimized builds <http://tensorflow.pypi.thoth-station.ninja>`_).
```

Comment:
@pacospace might be a good idea to generalize this to "`build_info`" so we have this info under same key also for other frameworks such as PyTorch.

Reply:
I agree! :beer: